File: 2-Patron/python/FabricaDeMonstruos.py
import FabricaDeCriaturas
import MonstruoColosal
import MonstruoComun
import GestorDeHabilidades
import GestorDeUltimate
import GestorDeResistencia


class FabricaDeMonstruos(FabricaDeCriaturas):
    def crearGrande(self, nombre):
        resistencia = GestorDeResistencia.generarResistencia()
        habilidad = GestorDeHabilidades().generarHabilidad()
        ultimate = GestorDeUltimate().generarUltimate()
        multiplicador_daño = 0.3
        return MonstruoColosal(nombre, 100, 1000, 1, habilidad, ultimate, resistencia, multiplicador_daño)

    def crearPequeña(self, nombre):
        resistencia = GestorDeResistencia.generarResistencia()
        habilidad = GestorDeHabilidades().generarHabilidad()
        ultimate = GestorDeUltimate().generarUltimate()
        return MonstruoComun(nombre, 100, 1000, 1, habilidad, ultimate, resistencia)

File: 2-Patron/php/FabricaDeMonstruos.php
<?php
class FabricaDeMonstruos
{
  function crearGrande($nombre, $nresistencia, $nhabilidad, $nultimate)
  {
    gResistencia = new GestorDeResistencia($nresistencia);
    gHabilidad = new GestorDeHabilidades($nhabilidad);
    gUltimate = new GestorDeUltimate($nultimate);

    $resistencia = gResistencia->generarResistencia();
    $habilidad = gHabilidad->generarHabilidad();
    $ultimate = gUltimate->generarUltimate();
    $multiplicador = 0.3;
    return  new MonstruoColosal($nombre, 100, 1000, 1, $habilidad, $ultimate, $resistencia, $multiplicador);
  }

  function crearPequena($nombre, $nresistencia, $nhabilidad, $nultimate)
  {
    gResistencia = new GestorDeResistencia($nresistencia);
    gHabilidad = new GestorDeHabilidades($nhabilidad);
    gUltimate = new GestorDeUltimate($nultimate);

    $resistencia = gResistencia->generarResistencia();
    $habilidad = gHabilidad->generarHabilidad();
    $ultimate = gUltimate->generarUltimate();
    $multiplicador = 0.3;
    return  new MonstruoComun($nombre, 100, 1000, 1, $habilidad, $ultimate, $resistencia, $multiplicador);
  }
}
?>
